test/e2e/gateway_api_test.go

```diff
@@ -276,6 +279,37 @@ func testGatewayAPIResourcesProtection(t *testing.T) {
 	}
 }
 
+// testOperatorDegradedCondition verifies that unmanaged (e.g. experimental)
+// Gateway API CRDs affect the ingress cluster operator's Degraded status.
+func testOperatorDegradedCondition(t *testing.T) {
+	// The experimental CRDs were created at the beginning of `TestGatewayAPI`.
+	// So the ingress cluster operator should report Degraded=True.
+	expectedDegraded := []configv1.ClusterOperatorStatusCondition{
+		{
+			Type:   configv1.OperatorDegraded,
+			Status: configv1.ConditionTrue,
+			Reason: "GatewayAPICRDsDegraded",
```

Comment:
This will result in false negatives in the case of `MultipleComponentsDegraded` where one of those components is the CRDs.

Reply:
I added [a check for `IngressNotDegraded` condition at the beginning of the test](https://github.com/openshift/cluster-ingress-operator/pull/1205/files#diff-0477a883800b78b4c8704dd53de74ed35c6a97a2c809d1581ceae78e10e9094fR424-R433), to avoid conflicts. The creation of the experimental CRDs was moved into this subtest making it less dependent of the main test.